tracer/src/Datadog.Trace/Configuration/TracerSettings.cs

```diff
@@ -398,20 +398,95 @@ internal TracerSettings(IConfigurationSource? source, IConfigurationTelemetry te
+
+            OtelMetricExportTimeoutMs = config
+                            .WithKeys(ConfigurationKeys.OpenTelemetry.MetricExportTimeoutMs)
+                            .AsInt32(defaultValue: 7_500);
+
+            OtlpMetricsProtocol = config
+                                 .WithKeys(ConfigurationKeys.OpenTelemetry.ExporterOtlpMetricsProtocol, ConfigurationKeys.OpenTelemetry.ExporterOtlpProtocol)
+                                 .GetAs(
+                                      defaultValue: new(OtlpProtocol.HttpProtobuf, "http/protobuf"),
+                                      converter: x => x switch
+                                      {
+                                          not null when string.Equals(x, "http/protobuf", StringComparison.OrdinalIgnoreCase) => OtlpProtocol.HttpProtobuf,
+                                          not null when string.Equals(x, "grpc", StringComparison.OrdinalIgnoreCase) => OtlpProtocol.Grpc,
+                                          not null when string.Equals(x, "http/json", StringComparison.OrdinalIgnoreCase) => OtlpProtocol.HttpJson,
+                                          _ => UnsupportedOtlpProtocol(inputValue: x ?? "null"),
+                                      },
+                                      validator: null);
+
+            var defaultUri = $"http://localhost:{(!OtlpMetricsProtocol.Equals(OtlpProtocol.Grpc) ? 4318 : 4317)}";
+            OtlpEndpoint = config
+                .WithKeys(ConfigurationKeys.OpenTelemetry.ExporterOtlpEndpoint)
+                .GetAs(
+                    defaultValue: new DefaultResult<Uri>(result: new Uri(defaultUri), telemetryValue: defaultUri),
+                    validator: null,
+                    converter: uriString => new Uri(uriString));
+
+            OtlpMetricsEndpoint = config
+                .WithKeys(ConfigurationKeys.OpenTelemetry.ExporterOtlpMetricsEndpoint)
+                .GetAs(
+                    defaultValue: new DefaultResult<Uri>(
+                        result: OtlpMetricsProtocol switch
+                        {
+                            OtlpProtocol.Grpc => OtlpEndpoint,
+                            _ => new Uri(OtlpEndpoint, "/v1/metrics")
+                        },
+                        telemetryValue: $"{defaultUri}/v1/metrics"),
+                    validator: null,
+                    converter: uriString => new Uri(uriString));
+
+            OtlpMetricsHeaders = config
+                            .WithKeys(ConfigurationKeys.OpenTelemetry.ExporterOtlpMetricsHeaders, ConfigurationKeys.OpenTelemetry.ExporterOtlpHeaders)
+                            .AsDictionaryResult(separator: '=')
+                            .WithDefault(new DefaultResult<IDictionary<string, string>>(new Dictionary<string, string>(), "[]"))
+                            .Where(kvp => !string.IsNullOrWhiteSpace(kvp.Key))
+                            .ToDictionary(kvp => kvp.Key.Trim(), kvp => kvp.Value?.Trim() ?? string.Empty);
```

Comment:
@andrewlock this what I end up coming up with, let me know if you think this is okay or should be improved.